snforge_std/src/cheatcodes/events.cairo

```diff
@@ -29,50 +20,73 @@ struct Event {
     data: Array<felt252>
 }
 
-
-/// An event spy structure, along with the events collected so far in the test.
-/// `events` are mutable and can be updated with `fetch_events`.
+/// An event spy structure containing an offset marking from which event given Spy
+/// should fetch events.
 #[derive(Drop, Serde)]
 struct EventSpy {
-    _id: felt252,
-    events: Array<(ContractAddress, Event)>,
+    _event_offset: usize
+}
+
+/// A wrapper structure on an array of events to handle filtering smoothly.
+#[derive(Drop, Serde)]
+struct Events {
+    events: Array<(ContractAddress, Event)>
 }
 
-trait EventFetcher {
-    /// Allows to update the structs' events field, from the spied contracts
-    fn fetch_events(ref self: EventSpy);
+trait EventSpyTrait {
+    /// Fetches all events accounting for the offset given [`EventSpy`] has.
```

Comment:
Method change from `fetch` to `get` while comment to `fetch`. Keep it consistent

Reply:
The [``] reference doesn't do anything as of now

Reply:
I also think the explanation is convoluted and too technical

Reply:
I'm having a dilemma right now, whether we should leave `fetch_events` as is, or go with `get_events`. There's also a `get_events` function in RPC - don't know if us naming it the same is good or bad. Waiting for your opinions.

Reply:
I would go with `fetch` as we are gaining events from backend

Reply:
> The [``] reference doesn't do anything as of now
I know, but I'm trying to be forward-compatible with cairo-doc

Reply:
I'm not sure it will be the format of tagging. You can probably leave it as-is tho